Source/Core/Core/HW/WiimoteReal/WiimoteReal.cpp

```diff
@@ -49,9 +47,49 @@ static std::mutex s_known_ids_mutex;
 
 std::mutex g_wiimotes_mutex;
 
+// Real wii remotes assigned to a particular slot.
 std::unique_ptr<Wiimote> g_wiimotes[MAX_BBMOTES];
+
+struct WiimotePoolEntry
+{
+  using Clock = std::chrono::steady_clock;
+
+  std::unique_ptr<Wiimote> wiimote;
+  Clock::time_point entry_time = Clock::now();
+
+  bool IsExpired() const
+  {
+    // Keep wii remotes in the pool for a bit before disconnecting them.
+    constexpr auto POOL_TIME = std::chrono::seconds{5};
+
+    return (Clock::now() - entry_time) > POOL_TIME;
+  }
+};
+
+// Connected wii remotes are placed here when no open slot is set to "Real".
+// They are then automatically disconnected after some time.
+std::vector<WiimotePoolEntry> g_wiimote_pool;
+
 WiimoteScanner g_wiimote_scanner;
 
+static void ProcessWiimotePool()
+{
+  std::lock_guard<std::mutex> wm_lk(g_wiimotes_mutex);
+
+  for (auto it = g_wiimote_pool.begin(); it != g_wiimote_pool.end();)
+  {
+    if (it->IsExpired())
+    {
+      NOTICE_LOG(WIIMOTE, "Removing expired wiimote pool entry.");
```

Comment:
Is this really a message that must always be visible in the log? If not, you should probably demote it to a very low level (like `INFO` or even `DEBUG`).
Reminder that `NOTICE` is "always visible" in Dolphin.

Reply:
I'll make it an `INFO`. I can never remember the difference.